tests/integration/test_interface.py

```diff
@@ -58,12 +58,18 @@ def test_generate_dataset_from_sample_and_source(dataset_name: str, config, tmpd
     """
     if "TODO" in dataset_name:
         pytest.skip(reason=dataset_name)
-
     data = request.getfixturevalue(f"sample_data_{dataset_name}")
-    source = _generate_non_sample_data_root(dataset_name, tmpdir, data)
     noised_sample = request.getfixturevalue(f"noised_sample_data_{dataset_name}")
+
+    # Split the sample dataset into two and save in tmpdir
+    outdir = tmpdir.mkdir(dataset_name)
+    split_idx = int(len(data) / 2)
+    data[:split_idx].to_parquet(outdir / f"{dataset_name}_1.parquet")
+    data[split_idx:].to_parquet(outdir / f"{dataset_name}_2.parquet")
     noising_function = DATASET_GENERATION_FUNCS.get(dataset_name)
-    noised_dataset = noising_function(seed=SEED, year=None, source=source, config=config)
+    noised_dataset = noising_function(seed=SEED, year=None, source=tmpdir, config=config)
```

Comment:
Just to be clear, we are calling each noise function a second time here. I don't think there's a way around this, but maybe we should be explicit that this is happening?

Reply:
I'll add comments to clarify this